sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala

```diff
@@ -46,4 +46,10 @@ class ExperimentalMethods private[sql]() {
 
   @volatile var extraOptimizations: Seq[Rule[LogicalPlan]] = Nil
 
+  override def clone(): ExperimentalMethods = {
```

Comment:
It sounds like we also need to add sync for both `extraStrategies ` and `extraOptimizations `